cropharvest/engineer.py

```diff
@@ -552,43 +614,79 @@ def create_h5_test_instances(
 
             # then, combine the instances into a test instance
             test_instance = TestInstance(np.stack(x), np.stack(y), np.stack(lats), np.stack(lons))
-            hf = h5py.File(self.test_savedir / f"{dataset}.h5", "w")
+            hf = h5py.File(TEST_FEATURES_FILEPATH / f"{dataset}.h5", "w")
             for key, val in test_instance.datasets.items():
                 hf.create_dataset(key, data=val)
             hf.close()
 
-    def create_h5_dataset(self, checkpoint: bool = True) -> None:
-        arrays_dir = self.savedir / "arrays"
-        arrays_dir.mkdir(exist_ok=True)
+    @staticmethod
+    def generate_bbox_from_paths(filepath: Path) -> Dict[Path, BBox]:
+        return {
+            p: BBox.from_eo_tif_file(p)
+            for p in tqdm(filepath.glob("**/*.tif"), desc="Generating BoundingBoxes from paths")
+        }
+
+    @staticmethod
+    def get_tif_paths(path_to_bbox, lat, lon, end_date, pbar):
+        candidate_paths = []
+        for p, bbox in path_to_bbox.items():
+            if bbox.contains(lat, lon) and fnmatch(p.stem, f"*dates=*_{end_date}*"):
+                candidate_paths.append(p)
+        pbar.update(1)
+        return candidate_paths
+
+    @classmethod
+    def match_labels_to_tifs(cls, labels: geopandas.GeoDataFrame) -> pd.Series:
+        bbox_for_labels = BBox(
+            min_lon=labels[RequiredColumns.LON].min(),
+            min_lat=labels[RequiredColumns.LAT].min(),
+            max_lon=labels[RequiredColumns.LON].max(),
+            max_lat=labels[RequiredColumns.LAT].max(),
+        )
+        # Get all tif paths and bboxes
+        path_to_bbox = {
+            p: bbox
+            for p, bbox in cls.generate_bbox_from_paths(EO_FILEPATH).items()
+            if bbox_for_labels.contains_bbox(bbox)
+        }
+
+        # Match labels to tif files
+        # Faster than going through bboxes
+        with tqdm(total=len(labels), desc="Matching labels to tif paths") as pbar:
+            tif_paths = np.vectorize(cls.get_tif_paths, otypes=[np.ndarray])(
+                path_to_bbox,
+                labels[RequiredColumns.LAT],
+                labels[RequiredColumns.LON],
+                labels[RequiredColumns.EXPORT_END_DATE],
+                pbar,
+            )
+        return tif_paths
+
+    def create_h5_dataset(self) -> None:
```

Comment:
What do you think about the type of comments in https://github.com/nasaharvest/crop-mask/blob/69da6cef8258b3171c6a02771bfc2219d8eadf5b/src/ETL/dataset.py#L320